### ISSUE TITLE
Implicit `todo` doesn't work on anonymous functions

### ISSUE THREAD

**Issue by digitalcora** [gleam-lang/gleam#2008]
Minimal example:

<table>
<tr><th>Before</th><th><code>gleam format</code></th></tr>

<tr>
<td><pre lang="gleam">pub fn main() {}</pre></td>
<td><pre lang="gleam">pub fn main() {
  todo
}</pre></td>
</tr>

<tr>
<td><pre lang="gleam">pub fn main() {
  fn(){}
}</pre></td>
<td><pre>error: Syntax error
  ┌─ &lt;stdin>:9:1
  │
9 │ }
  │ ^ I was not expecting this.
Expected one of: 
The body of a function</pre></td>
</tr>
</table>

Basically, a named function with an empty body is treated as though the body is `todo`, but an anonymous function with an empty body is a syntax error.

From what I can see, the `todo` inserting feature was added in #1688.

**Comment by lpil** [comment]
Hello! Yes, anonymous functions must have a body by design. Perhaps we could have implicit todo insertion here also? I am unsure what would be best, I'm interested in what folks think.

**Comment by digitalcora** [comment]
I do like my tables 😄

Personally I think 1) it would be nice to have it work consistently between named and anonymous functions, so you don't have to remember that one works differently from the other; and 2) of the two possible decisions there, I'd prefer having implicit `todo`.

For context, this came up while I was writing a `fold` call that started going over the line length, so I finished typing `with: fn() {}` and hit Format in my editor, expecting the formatter to be fine with this, as it is for named functions. Based on experience with other languages, I think it's unexpected to have to explicitly add something to a function body just to be able to format the code; having an empty function body is sometimes a _warning_, but not an error.

**Comment by lpil** [comment]
That seems very reasonable. I'm convinced, let's add this here also.
